op-devstack/dsl/ecotone_fees.go

```diff
@@ -58,24 +58,27 @@ func (ef *EcotoneFees) ValidateTransaction(from *EOA, to *EOA, amount *big.Int)
 	vaultsAfter := ef.getVaultBalances(client)
 	vaultIncreases := ef.calculateVaultIncreases(vaultsBefore, vaultsAfter)
 
-	l1Fee := big.NewInt(0)
-	if receipt.L1Fee != nil {
-		l1Fee = receipt.L1Fee
-	}
+	// In Ecotone, L1 fee includes both base fee and blob base fee components
+	l1Fee := vaultIncreases.L1FeeVault // Use actual vault increase as the source of truth
 
-	block, err := client.InfoByHash(ef.ctx, receipt.BlockHash)
-	ef.require.NoError(err)
+	// Calculate L2 fees
+	baseFee := vaultIncreases.BaseFeeVault       // Use actual vault increase as the source of truth
+	priorityFee := vaultIncreases.SequencerVault // Use actual vault increase as the source of truth
+	l2Fee := new(big.Int).Add(baseFee, priorityFee)
 
-	baseFee := new(big.Int).Mul(block.BaseFee(), big.NewInt(int64(receipt.GasUsed)))
-	l2Fee := new(big.Int).Mul(receipt.EffectiveGasPrice, big.NewInt(int64(receipt.GasUsed)))
```

Comment:
Also this, we need to preserve the checks for the receipt included fee data like `receipt.EffectiveGasPrice`.

Reply:
Hopefully addressed in [e0feddc](https://github.com/ethereum-optimism/optimism/pull/16817/commits/e0feddc48d33b660574f9e8bf6a96b4cf35c91b3). Please take a look and let me know if this addresses your feedback.